enterprise/com/src/main/java/org/neo4j/com/storecopy/FileMoveProvider.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.com.storecopy;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.impl.store.StoreType;
+
+import static org.neo4j.helpers.collection.Iterables.asList;
+
+public class FileMoveProvider
+{
+
+    private final FileMoveActionInformer fileMoveActionInformer;
+    private final PageCache pageCache;
+
+    public FileMoveProvider( PageCache pageCache )
+    {
+        this( pageCache, StoreType::shouldBeManagedByPageCache );
+    }
+
+    public FileMoveProvider( PageCache pageCache, FileMoveActionInformer fileMoveActionInformer )
+    {
+        this.pageCache = pageCache;
+        this.fileMoveActionInformer = fileMoveActionInformer;
+    }
+
+    /**
+     * Construct a stream of files that are to be moved
+     *
+     * @param dir the source location of the move action
+     * @return a stream of the entire contents of the source location that can be applied to a target location to perform a move
+     */
+    public Stream<FileMoveAction> traverseGenerateMoveActions( File dir )
+    {
+        return traverseGenerateMoveActions( dir, dir );
+    }
+
+    /**
+     * Copies <b>the contents</b> from the directory to the base target path.
+     * <p>
+     * This is confusing, best read the tests
```

Comment:
Maybe include an example of usage in Javadoc?

Reply:
Addressed